File: Sacred-Ecommerce-Microservice/carts/src/main/java/com/sacred/carts/exceptions/ResourceNotFoundException.java
package com.sacred.carts.exceptions;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.NOT_FOUND)
public class ResourceNotFoundException extends RuntimeException {

    public ResourceNotFoundException(String resourceName, String fieldName, String value){
        super(String.format("%s not found by  %s with given value %s", resourceName, fieldName, value));
    }

}

File: Sacred-Ecommerce-Microservice/orderManagement/src/main/kotlin/com/sacred/ordermanagement/exceptions/ResourceNotFoundException.kt
package com.sacred.ordermanagement.exceptions

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(value = HttpStatus.NOT_FOUND)
class ResourceNotFoundException(
        private val resourceName: String,
        private val fieldName: String,
        private val value: String
): RuntimeException("$resourceName not found by $fieldName with given value $value")